### ISSUE TITLE
Maid Paradise Mezase! Maid Number One! fps drop

### ISSUE THREAD

**Issue by IrfanH495** [hrydgard/ppsspp#18452]
### Game or games this happens in

NPJH00099 - メイド☆ぱらだいす ～目指せ！メイドナンバーワン！～

### What area of the game

while playing quiz and giving massages

### Speed seen in PPSSPP

78% 24/30

### GE frame capture and debug statistics

[NPJH00099_0001.zip](https://github.com/hrydgard/ppsspp/files/13503818/NPJH00099_0001.zip)
![Screenshot_20231130-003021_PPSSPP](https://github.com/hrydgard/ppsspp/assets/78854859/9b8a2d1c-d490-454d-89b7-03c9dec9ca00)
![Screenshot_20231130-001532_PPSSPP](https://github.com/hrydgard/ppsspp/assets/78854859/a397b49f-99ab-4ee6-886b-c517bea70af5)

### Platform

Android

### Mobile phone model or graphics card

Samsung M20

### PPSSPP version affected

ppsspp-v1.16.6-368-g737ec3e90-

### Last working version

.

### Graphics backend (3D API)

Vulkan

### Any other notes or things you've tried

I tried adding to BlockTransferAllowCreateFB make damage in the massaging part right where the character's image is talking
![Screenshot_20231130-001909_PPSSPP](https://github.com/hrydgard/ppsspp/assets/78854859/2dac806c-db5b-4740-8751-0421cbc2253d)

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try resetting settings or older versions and include if the issue is related.
- [X] Try changing graphics settings to determine if one causes the slowdown.
- [X] Include logs or screenshots of issue.

**Comment by hrydgard** [comment]
Strange game haha.

It's clear that the performance issue is the readback. Reducing render resolution to 1x can potentially help a bit.

But this seems likely (although absolutely not certain!) to be a case where reading back slightly old data would work fine, unlike some other games that need readbacks. We have a mode that allows reading back data from a few frames ago that runs faster, although it's not exposed. I think I'll add a compat flag for it so you can play with it and see if it helps.

**Comment by hrydgard** [comment]
Created https://github.com/hrydgard/ppsspp/pull/18461 .

**Comment by IrfanH495** [comment]
> Strange game haha.

yes unlike the main game, this is just a collection of mini games from GachiTora!  Abarenbou Kyoushi in High School game



> It's clear that the performance issue is the readback. Reducing render resolution to 1x can potentially help a bit.

in the quiz place there is no fps drop on 1x PSP, in the massage place 2x PSP there is no fps drop

**Comment by hrydgard** [comment]
Alright. Anyway, the setting is in, so once there's a build, feel free to see if it helps. It may also very well break things!

**Comment by hrydgard** [comment]
Is that with Vulkan?

**Comment by IrfanH495** [comment]
> Is that with Vulkan?

yes
This is also the same as Gachitora after finding out he also has readback, didn't realize I was playing it 100% finished but used compact to increase speed
![Screenshot_20231202-194922_PPSSPP](https://github.com/hrydgard/ppsspp/assets/78854859/fe2ada50-d375-4a0f-8e94-a7ac629d4a1f)
![Screenshot_20231202-194358_PPSSPP](https://github.com/hrydgard/ppsspp/assets/78854859/ea12955d-92d7-4f55-b25a-65cce6ed1161)
![Screenshot_20231202-194125_PPSSPP](https://github.com/hrydgard/ppsspp/assets/78854859/b6d4d169-4a73-4bf7-bd8d-f8c53dd59699)

**Comment by hrydgard** [comment]
Alright, I guess the setting won't work with this game then.
